src/controllers/CategoryFacetQueryController.ts

```diff
@@ -11,19 +11,8 @@ export class CategoryFacetQueryController {
 
   public putCategoryFacetInQueryBuilder(queryBuilder: QueryBuilder, path: string[], maximumNumberOfValues: number): number {
     const positionInQuery = queryBuilder.categoryFacets.length;
-    if (path.length != 0) {
-      queryBuilder.advancedExpression.addFieldExpression(this.categoryFacet.options.field as string, '==', [
-        path.join(this.categoryFacet.options.delimitingCharacter)
-      ]);
-    }
-    const categoryFacetsRequest: ICategoryFacetRequest = {
-      field: this.categoryFacet.options.field as string,
-      path,
-      injectionDepth: this.categoryFacet.options.injectionDepth,
-      maximumNumberOfValues,
-      delimitingCharacter: this.categoryFacet.options.delimitingCharacter
-    };
-    queryBuilder.categoryFacets.push(categoryFacetsRequest);
+    this.addQueryFilter(queryBuilder, path);
```

Comment:
When there is a conditional in the implementation, I like to mention it in the name so that when reading the code I do not assume the action is performed regardless. e.g. `addQueryFilterIfNeeded`